pkg/bucketeer/api/api.go

```diff
@@ -0,0 +1,244 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/ca-dp/bucketeer-go-server-sdk/pkg/bucketeer/user"
+)
+
+const (
+	evaluationAPI    = "/get_evaluation"
+	eventsAPI        = "/register_events"
+	authorizationKey = "authorization"
+)
+
+type EventType string
+
+type metricsDetailEventType string
+
+//nolint:lll
+const (
+	GetEvaluationLatencyMetricsEventType metricsDetailEventType = "type.googleapis.com/bucketeer.event.client.GetEvaluationLatencyMetricsEvent"
+	GetEvaluationSizeMetricsEventType    metricsDetailEventType = "type.googleapis.com/bucketeer.event.client.GetEvaluationSizeMetricsEvent"
+	TimeoutErrorCountMetricsEventType    metricsDetailEventType = "type.googleapis.com/bucketeer.event.client.TimeoutErrorCountMetricsEvent"
+	InternalErrorCountMetricsEventType   metricsDetailEventType = "type.googleapis.com/bucketeer.event.client.InternalErrorCountMetricsEvent"
+)
+
+const (
+	GoalEventType       EventType = "type.googleapis.com/bucketeer.event.client.GoalEvent"
+	EvaluationEventType EventType = "type.googleapis.com/bucketeer.event.client.EvaluationEvent"
+	MetricsEventType    EventType = "type.googleapis.com/bucketeer.event.client.MetricsEvent"
+)
+
+type wellKnownTypes string
+
+const DurationType wellKnownTypes = "type.googleapis.com/google.protobuf.Duration"
+
+type SourceIDType int32
+
+const (
+	SourceIDGoServer SourceIDType = 5
+)
+
+type RegisterEventsRequest struct {
+	Events []*Event `json:"events,omitempty"`
+}
+
+type registerEventsRequest struct {
+	*RegisterEventsRequest
+}
+
+type RegisterEventsResponse struct {
+	Errors map[string]*RegisterEventsResponseError `json:"errors,omitempty"`
+}
+
+type GetEvaluationRequest struct {
+	Tag       string     `json:"tag,omitempty"`
+	User      *user.User `json:"user,omitempty"`
+	FeatureID string     `json:"featureId,omitempty"`
+}
+
+type getEvaluationRequest struct {
+	*GetEvaluationRequest
+	SourceID SourceIDType `json:"sourceId,omitempty"`
+}
+
+type GetEvaluationResponse struct {
+	Evaluation *Evaluation `json:"evaluation,omitempty"`
+}
+
+type Event struct {
```

Comment:
All the structs in the `api.go` are data models. They should be in another pkg to also avoid future cycle dependencies.
Before we changed it from gRPC to REST, we had all data objects in the `pkg/proto` package.
It will be better to put those structs in the `pkg/model`.

E.g.
https://github.com/bucketeer-io/android-client-sdk/tree/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/model

https://github.com/ca-dp/bucketeer-ios-sdk/tree/v2/Bucketeer/Sources/Internal/Model

https://github.com/ca-dp/bucketeer-web-sdk/tree/master/src/objects